### PR TITLE
add Bid module and tests

### PR DESCRIPTION
**Pull request by martinstannard**
Adds a Bid module for testing whether a bid is valid or not

### CODE REVIEW

**Review by wiserfirst — APPROVED** [review]
🚢 💯

File: apps/game/lib/game/bid.ex [hunk 1]
```diff
@@ -0,0 +1,43 @@
+defmodule Game.Bid do
+
+  defstruct dice_count: 1, face_value: 2
+
+  alias Game.Bid
+
+  @spec valid?(%Bid{}, %Bid{}, integer) :: boolean
+  def valid?(new_bid, current_bid, dice_count) do
+    has_valid_face_value?(new_bid) &&
```
Comment (JulianLeviston) [inline apps/game/lib/game/bid.ex:9]: How about...
```
Enum.all?([
    has_valid_face_value?(new_bid),
    has_valid_dice_count?(new_bid, dice_count),
    new_bid_is_higher?(new_bid, current_bid)
])
```

**Comment by JulianLeviston** [comment]
Nice!

File: apps/game/lib/game/bid.ex (hunk 1 above)
Comment (martinstannard) [reply apps/game/lib/game/bid.ex:9]: I think the `&&` version will bail at the first failure, whereas the Enum would have to calculate them all first?

**Pull request merged**